runtime-sdk/modules/evm/src/evm_backend.rs

```diff
@@ -166,82 +148,45 @@ impl<'c, C: oasis_runtime_sdk::Context> ApplyBackend for Backend<'c, C> {
+                        //       existed.
+                    }
 
-                        for (index, value) in storage {
-                            let idx: H256 = index.into();
-                            let val: H256 = value.into();
+                    // Handle storage updates.
+                    for (index, value) in storage {
+                        let idx: H256 = index.into();
+                        let val: H256 = value.into();
 
-                            let state = self.ctx.get_mut().runtime_state();
-                            let s_store = storage::PrefixStore::new(state, &crate::MODULE_NAME);
-                            let storages = storage::PrefixStore::new(s_store, &state::STORAGES);
-                            let mut s = storage::TypedStore::new(storage::HashedStore::<
-                                _,
-                                blake3::Hasher,
-                            >::new(
-                                storage::PrefixStore::new(storages, &addr),
-                            ));
-                            if value == primitive_types::H256::default() {
-                                s.remove(&idx);
-                            } else {
-                                s.insert(&idx, val);
-                            }
+                        let mut store = state::storage(self.ctx.get_mut().runtime_state(), &addr);
+                        if value == primitive_types::H256::default() {
+                            store.remove(&idx);
+                        } else {
+                            store.insert(&idx, val);
                         }
-
-                        account.balance == primitive_types::U256::zero().into()
-                            && account.nonce == primitive_types::U256::zero().into()
-                    };
-
-                    if is_empty && delete_empty {
-                        let state = self.ctx.get_mut().runtime_state();
-                        let a2_store = storage::PrefixStore::new(state, &crate::MODULE_NAME);
-                        let mut accounts = storage::TypedStore::new(storage::PrefixStore::new(
-                            a2_store,
-                            &state::ACCOUNTS,
-                        ));
-
-                        accounts.remove(&addr);
                     }
                 }
-                Apply::Delete { address } => {
-                    let addr: H160 = address.into();
-                    let state = self.ctx.get_mut().runtime_state();
-                    let store = storage::PrefixStore::new(state, &crate::MODULE_NAME);
-                    let mut accounts = storage::TypedStore::new(storage::PrefixStore::new(
-                        store,
-                        &state::ACCOUNTS,
-                    ));
-
-                    accounts.remove(&addr);
+                Apply::Delete { .. } => {
+                    // Accounts cannot be deleted.
```

Comment:
what from evm would attempt to do this?

Reply:
The dumb selfdestruct opcode (see my comment above).